src/domain/utils.ts

```diff
@@ -0,0 +1,7 @@
+import { RepositoryError, UnknownRepositoryError } from "./errors"
+
+export const isRepoError = (obj): boolean =>
+  obj instanceof RepositoryError || obj instanceof UnknownRepositoryError
+
+export const isWalletInvoice = (invoice): invoice is WalletInvoice =>
+  (invoice as WalletInvoice).paymentHash !== undefined
```

Comment:
This should not be needed! The type is defined as:
```
type WalletInvoice = {
  paymentHash: PaymentHash
  ...
}
```

We should always assume that when we have a `type` it is valid like the definition. If there is a way we can have a `WalletInvoice` that doesn't have a `paymentHash` then this is a bug and needs to be fixed at the root cause. Adding this method is a hack that masks the root cause. The whole point of using typescript is to avoid these noisy checks.

Reply:
I know but is the technique suggested to let TS compiler know/infer the type (after validation), I guess we will not need it because the error extend changes but maybe will be useful later... the syntax is `isWalletInvoice = (invoice): invoice is WalletInvoice =>` (check the weird return syntax)  but you need to check a property inside the method